presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -285,12 +285,13 @@ static void validateFile(
                     new StorageStripeMetadataSource(),
                     Optional.of(writeValidation),
                     orcReaderOptions,
-                    false);
+                    false,
+                    new NoopOrcAggregatedMemoryContext());
             try (OrcBatchRecordReader orcRecordReader = orcReader.createBatchRecordReader(
                     readTypes.build(),
                     OrcPredicate.TRUE,
                     hiveStorageTimeZone,
-                    newSimpleAggregatedMemoryContext(),
+                    new NoopOrcAggregatedMemoryContext(),
```

Comment:
make a static final variable in class `NoopOrcAggregatedMemoryContext` so we don't have to "new" it every time.

Reply:
I am not sure if Noop is a valid argument here. Because its not equivalent of ``newSimpleAggregatedMemoryContext()``. Should we instead use the ``OrcAggregatedMemoryContext `` sent to ``OrcReader`` ?

Also looks like tests in ``TestOrcReaderMemoryUsage`` expect a valid memory context rather than a noop one as they assert on ``reader.getCurrentStripeRetainedSizeInBytes()``.  Let me know your comments.

Reply:
You need to pass the HiveOrc memory context rather than `NOOP_ORC_AGGREGATED_MEMORY_CONTEXT` in the test right?

Reply:
This test is in presto-orc module

Reply:
That is OK, presto-orc can depend on anything (e.g., `presto-main`) as long as they are in test scope. In this case, (I guess we cannot even depend on presto-hive for test scope because we will have circular dependency), we can either
- Move the test to presto-hive, or
- (1) copy HiveOrc..Context to `presto-orc` as `TestingHiveOrc..Context` and use that and (2) make `presto-orc` depending on `presto-memory-context` with test scope.